qlib/kernel/loader/vdso.rs

```diff
@@ -39,6 +42,83 @@ pub struct VdsoInternal {
     pub vdsoAddr: u64,
     pub vdsoLen: usize,
     pub phdrs: Vec<ProgramHeader64>,
+    pub vdso_symbols: Option<VdsoArchSymbols>,
+}
+
+struct VdsoSymbol {
+    name:&'static str,
+    page_offset: Option<u64>,
+}
+
+/// Bookkeep the exported symbols for x86_64.
+#[cfg(target_arch = "x86_64")]
+pub struct VdsoArchSymbols {
+}
+
+#[cfg(target_arch = "x86_64")]
+impl VdsoArchSymbols {
+    pub fn set_symbols_page_offset(&mut self, elf_file: &ElfFile, symbol_table: &[Entry64]) {
```

Comment:
The code is assuming that vdso is one single page right? Perhaps add a memory limit in the linker script to make sure of that?